### DISCUSSION TITLE
PylonCapture + OpticalFlow node issue

### DISCUSSION THREAD

**Discussion by thedennislab** [bonsai-rx/bonsai#873] [Q&A] [answered]
hi folks,

I'm using a Basler camera and the PylonCapture node to retrieve frames. This looks great. Separately, I can use my desktop camera with CameraCapture node and an OpticalFlow node and get nice information from it. However, when I try to use the PylonCapture node with the OpticalFlow node, regardless of settings, the PylonCapture node produces images that are looping the first few seconds of video very slowly, except for a tiny band at the top (horizontal stripe at the very top of the image) which reflects the current camera info.

My guess is this is somehow a memory allocation issue, but that's just a guess.

"Go To Definition" takes me to something a little too high level to understand how these things are working and allocating memory. So either any advice on how to solve this issue or guidance on how to delve into the code for a given node would be helpful (related, some docs about this that are easy to find would be great for the site)

thanks all!

**Comment by emilyjanedennis** [comment, marked as answer]
we solved this issue by downsizing the images before passing them to optical flow.
